### PR TITLE
Post page volunteer

### PR DESCRIPTION
**Pull request by knargocer**
PostPageVolunteer

Post page volunteer:
I wanted to chenge my branchs name because it was PostPages I thought it would be better to have two branches for two diffrent tasks. Thus on the process deleted my previous one accidentally. 

 * PostList contains the posts to be shown
* Post is the single posts that are created by the volunteers.
 * PostHeader contains the picture and the writing of the page header

I have to make the unit tests also the page needs to reload after performing search and filter operations

### CODE REVIEW

**Comment by Ammar-64** [comment]
@knargocer Can you also merge dev here and make sure all checks are passing, so I can review it

File: src/components/PostListVolunteer/index.css
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.css]: Can you give a className for the label so it doesn’t affect all the labels in the site

File: src/components/PostListVolunteer/index.css
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.css]: Can you give more specific name for the search here, i.e. postSearch

File: src/components/PostListVolunteer/index.css
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.css]: What does first select means?

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: What is secPost? Can it be more meaningful and avoid abbreviations

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: Can names be more meaningful here, i.e. searchKeyWord instead of search, and searchResults instead of results

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: All texts should be coming from i18next, including All

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: The placeHolder should be coming from i18next too

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: Instead of loading you can add some gif image that indicates loading, or you can keep it up to you, this is just a suggestion

File: src/components/PostPageVolunteerHeader/index.css
Comment (Ammar-64) [inline src/components/PostPageVolunteerHeader/index.css]: img here will affect all the images on the site, make sure to give a className instead

File: src/components/PostVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostVolunteer/index.jsx]: The alt text should be coming from i18next as well

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid, but please take the comments I added into consideration

**Comment by knargocer** [comment]
@Ammar-64 PTAL

File: src/components/PostListVolunteer/index.css
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.css]: Ola has added the font family to app.css which means you don’t need to do it every time, so you can just delete font families from here

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: You don’t need to import i18n here it’s already imported in the main index.jsx

File: src/components/PostListVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostListVolunteer/index.jsx]: I wouldn’t allow full screen

File: src/components/PostPageVolunteerHeader/index.css
Comment (Ammar-64) [inline src/components/PostPageVolunteerHeader/index.css]: Same for font family here

File: src/components/PostVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostVolunteer/index.jsx]: You can use conditional rendering here instead of ternary operator i.e   `{postInfo.postInfo && postInfo.postInfo.title}`

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Some extra few changes and if you can remove package-lock.json, this comes with npm and we’re using yarn

**Comment by knargocer** [comment]
@Ammar-64 PTAL

File: src/App.jsx
Comment (Luey-A) [inline src/App.jsx]: I think commented lines should be removed from the code.

File: src/components/PostPageVolunteerHeader/index.css
Comment (Luey-A) [inline src/components/PostPageVolunteerHeader/index.css]: i think this is unnecessary, even if it was removed it would give the same result because it is auto

File: src/components/PostPageVolunteerHeader/index.css
Comment (Luey-A) [inline src/components/PostPageVolunteerHeader/index.css:4]: Ola added this to the App.css so it is implemented for all the code so there is not need to mention it in each css file

File: src/components/PostVolunteer/index.css
Comment (Luey-A) [inline src/components/PostVolunteer/index.css]: we agreed to use em

File: src/components/PostVolunteer/index.css
Comment (Luey-A) [inline src/components/PostVolunteer/index.css]: em again

File: src/components/PostListVolunteer/index.css
Comment (Luey-A) [inline src/components/PostListVolunteer/index.css]: classNames are not very clear, it can be changed to something more meaningful. this includes filter sort title and almost all one word css

File: src/components/PostVolunteer/index.css
Comment (Luey-A) [inline src/components/PostVolunteer/index.css]: all pxs need to be changed to em

File: src/i18n.js
Comment (Luey-A) [inline src/i18n.js]: should be removed

**Review by Luey-A — CHANGES_REQUESTED** [review]

File: src/components/PostPageVolunteerHeader/index.css
Comment (knargocer) [reply src/components/PostPageVolunteerHeader/index.css:4]: When I removed it, it made the font change therefore I kept it but deleted the import

File: src/i18n.js
Comment (knargocer) [reply src/i18n.js]: I added it because there was an error for importing common_en, common_ar and common_tr

**Comment by knargocer** [comment]
@Luey-A  PTAL

**Comment by orcunsagirsoy** [comment]
-The comment I made on the receiver version of this page + could you please fix the height of the cards, they are different right now.
-Some city names doesn't fit with its filtering container border, maybe you can consider making them bigger (i.e. when you filter by Istanbul)

File: src/components/PostVolunteer/index.jsx
Comment (Ammar-64) [inline src/components/PostVolunteer/index.jsx]: What does this return, does it make sense to return loading, until the page loads?

**Review by Ammar-64 — COMMENTED** [review]
Solid, only one comment, I need your insight on

File: src/components/PostVolunteer/index.jsx
Comment (knargocer) [reply src/components/PostVolunteer/index.jsx]: I added it because when there was only if and no else it gave an error that the function does not return anything. I think changing it to loading gif makes more sense

**Comment by Luey-A** [comment]
You have one conflicting file src/App.jsx that need to be resolved.

**Comment by Luey-A** [comment]
> @Luey-A PTAL

Thank you for the changes, it is great now.

File: src/components/Post/index.css
Comment (Ammar-64) [inline src/components/Post/index.css]: Use hex colors instead of rgb please,

File: src/components/Post/index.css
Comment (Ammar-64) [inline src/components/Post/index.css]: Em instead of px here

File: src/components/PostList/index.jsx
Comment (Ammar-64) [inline src/components/PostList/index.jsx]: Delete the console.log it shouldn’t be in production

File: src/components/PostList/index.jsx
Comment (Ammar-64) [inline src/components/PostList/index.jsx]: e doesn’t make sense as a variable name

File: src/containers/PostPageVolunteer/index.jsx
Comment (Ammar-64) [inline src/containers/PostPageVolunteer/index.jsx:9]: What about the receiver post, where are you passing the props to them?

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Can you take a look at the comments, I’m still confused where are you passing the receiver post

File: src/containers/PostPageVolunteer/index.jsx
Comment (knargocer) [reply src/containers/PostPageVolunteer/index.jsx:9]: It is another container <PostPageReciever>.In that container the same way, I am passing the props as "recieverPost".

**Comment by knargocer** [comment]
@Ammar-64  PTAL

File: src/containers/PostPageVolunteer/index.jsx
Comment (Ammar-64) [reply src/containers/PostPageVolunteer/index.jsx:9]: Cool, I see the. Wait until this merged then pull for the other request, instead of add the whole code again

**Review by Ammar-64 — APPROVED** [review]
LGTM

**Review by Luey-A — APPROVED** [review]

**Pull request merged**
